### ISSUE TITLE
sql: incomplete vectorized traces

### ISSUE THREAD

**Issue by glennfawcett** [cockroachdb/cockroach#59555]
**Describe the problem**

I ran across an interesting observation while trying to explain how queries are run in parallel within CRDB.  I was using the internal KV workload and have a fair number of ranges distributed across three nodes:

```
root@:26257/kv> SELECT k2, count(*) FROM kv GROUP BY k2;
    k2   |  count
---------+----------
  z      | 4561620
  needle |       8
  
```  
I was expecting that the above query would run in parallel, but when looking at the Jaeger trace collected, it appears to not be serial.  

<img width="1567" alt="Screen Shot 2021-01-19 at 11 18 57 AM" src="https://user-images.githubusercontent.com/5796677/106195748-6c35fa00-6165-11eb-9f1e-b9121e3ae643.png">
[stmt-bundle-625911811735650305.zip](https://github.com/cockroachdb/cockroach/files/5889843/stmt-bundle-625911811735650305.zip)
[stmt-bundle-628463749489590273.zip](https://github.com/cockroachdb/cockroach/files/5889847/stmt-bundle-628463749489590273.zip)

At the advise of engineering, I ran this by disabling vectorization `SET vectorize = off` and it appeared to run as expected albeit longer.

**Comment by RaduBerinde** [comment]
To clarify - the query runs in parallel on the three nodes, but the traces for the actual flow and scans are missing from node 2 and 3. It looks like a problem with tracing when using the vectorized engine. I tried to repro locally in a simpler case and could not - I saw traces for scans from all nodes.

**Comment by yuzefovich** [comment]
Yeah, it is a known issue (#55821) that we intend to address for 21.1 release.

**Comment by yuzefovich** [comment]
An update here is that #61380 will fix it and will be in 21.1:
![Screen Shot 2021-03-03 at 7 11 27 PM](https://user-images.githubusercontent.com/5135385/109905839-431cf380-7c54-11eb-901a-10f99f84d3a4.png)
